doc/api/cli.md

```diff
@@ -550,7 +551,11 @@ changes:
                  `--experimental-loader`.
 -->
 
-Specify the `module` of a custom experimental [ECMAScript module loader][].
+> This flag is discouraged and may be removed in a future version of Node.js.
+> Please use
+> [`--import` with `register()`][module customization hooks: enabling] instead.
```

Comment:
This doesn’t feel like an improvement. We don’t need to explain why it’s discouraged. It may be “less robust” but it’s shorter than `--import` with a `data:` URL, so there are tradeoffs. We don’t need to get into it in the docs.

Reply:
Perhaps, but I don’t think your suggestion really answers the “why” question, and to do so properly would require more space than is worth. I think the documentation just needs to explain how to use the APIs, not get into digressions about why decisions were made or current designs are better than previous ones. If someone truly wants to know then they’ll search for it and find GitHub issues explaining it.